lib/url.js

```diff
@@ -563,7 +563,7 @@ Url.prototype.format = function() {
   }
 
   var protocol = this.protocol || '';
-  var pathname = this.pathname || '';
+  var pathname = (this.pathname + '') || '';
```

Comment:
Now it's a breaking change. if `this.pathname === null`, `pathname` will be the `'null'` string.
Something like

```js
var pathname = (this.pathname || '').toString();
```

should work.